Tests/Realm.Tests/Sync/SyncConfigurationTests.cs

```diff
@@ -79,6 +80,13 @@ public void SyncConfiguration_WithAbsolutePath()
             Assert.That(config.DatabasePath, Is.EqualTo(path));
         }
 
+        [Test]
+        public void FlexibleSyncConfiguration_throws_When_Assigned_DiscardLocalResetHandler()
+        {
+            var conf = GetFakeFLXConfig();
+            Assert.That(() => { conf.ClientResetHandler = new DiscardLocalResetHandler(); }, Throws.TypeOf<NotSupportedException>());
+        }
+
```

Comment:
Needs a test that validates that `DiscardLocal` is the default client reset strategy.

Reply:
As a note I put the test in the `SessionTests.cs` to keep all testing regarding this new client reset API in one place.